### ISSUE TITLE
Export GBIX to PNG

### ISSUE THREAD

**Issue by barquetin16** [xdanieldzd/Scarlet#10]
Hi can you please explain me the correct way to convert a .gbix(uvr) to png from the game K-on Houkago Live, An if just in case you know how convert the .uno files from the game to any readable 3d format.

**Comment by lfdaryl** [comment]
the vertex position and uv (in .uno) can be gained form hex editor in float mode, but all Sega pvr tools found in network didn't work correctly when I tried to open uvr files. The data formats of textures are 0xa8 or 0x88,different from normal gbix(pvr), while those tools only read 0x00-0x12 formats. https://github.com/nickworonekin/puyotools/wiki/PVR-Texture

**Comment by barquetin16** [comment]
Thanks for the information, so there is no way that Scarlet convert the .uvr gbix file like it says on the readme, because it says K-on as example.

**Comment by xdanieldzd** [comment]
Scarlet _can_ convert GBIX format images (*.uvr) to PNG:

![temp682](https://user-images.githubusercontent.com/7199225/33858800-719d54d0-ded1-11e7-9d2f-3e34c87bced5.png)

It cannot convert PNGs back into GBIX, it might not work with every GBIX file - you can see some garbled results in the screenshot, which may or may not have been fixed since these were converted (screenshot shows old testing files, I'll need to re-check if they all convert properly) - but it should work just fine for many of them.

(Also, sorry for the late reply, real life things getting in the way here and there.)

**Comment by barquetin16** [comment]
Hi thanks for your response can you tell me what the code line will be to convert .uvr to .png, i cant get it working.

**Comment by xdanieldzd** [comment]
It should work fine if you just drag and drop your .uvr files (or FMDX .bin archives, containing .uvr files among others) onto ScarletTestApp.exe. To be sure, I just downloaded [the most recent release, v1.1.4],(https://github.com/xdanieldzd/Scarlet/releases/tag/v1.1.4) and tried it with a .uvr file:

![temp684](https://user-images.githubusercontent.com/7199225/33860316-864550a2-ded8-11e7-8a52-5d911b600a10.png)

You should see something along those lines as well. If not and you're not getting any PNGs, please post a screenshot of the converter's output.

**Comment by barquetin16** [comment]
Here is a screenshot from the converter
![scarlet](https://user-images.githubusercontent.com/19699909/33860644-cc8661e0-dea7-11e7-870a-66b4452b5569.png)

**Comment by xdanieldzd** [comment]
Ahh, I think I know what's wrong, that file (and probably any others you've tried that give the same error) aren't supported yet. More specifically, their image formats haven't been verified and added to Scarlet's GBIX handler yet, in which case it throws the exception you're seeing.

I'll see what I can do over the next couple of days or so, time permitting. Should hopefully not be _too_ hard to fix these issues, will just have to find the ISO again or, worst case, dump my copy of the game again.

**Comment by lfdaryl** [comment]
I could found coordinates of vertexes in .uno and export them (just used Excel hhhh), but no hint of  the face. It looks a bit hard to fix the model because of my poor skill of coding and modeling.
Here's a screenshot of Hirasawa Yui
![image](https://user-images.githubusercontent.com/34444412/33887624-5168465c-df85-11e7-89af-b6ea2f7db486.png)
I found some blocked structures in .uno (usually the size is 28 bytes, sometimes 24 bytes or 20 bytes)  
![image](https://user-images.githubusercontent.com/34444412/33887876-3ffda898-df86-11e7-8fae-8548ae86335e.png)
Obviously the last three numbers are vertex coordinates, the third and the forth are probably UV coordinates, but I don't know the true meaning of the first two. 
[YUI_Body.txt](https://github.com/xdanieldzd/Scarlet/files/1551806/YUI_Body.txt)
（a sample of the model）
I'd like to ask if anyone can get the information of faces.

**Comment by xdanieldzd** [comment]
GBIX support has been improved quite a bit with commit 600ab4c, most files should convert properly now. There's still a few issues I'd like to try and figure out before posting a new release build, tho.

As for .uno models, I might take a look at them as well, but I won't promise anything. I might figure out enough of them to code a viewer or dumper, but I might also not be able to do much of anything with them, we'll see.

**Comment by barquetin16** [comment]
Thanks, This guy just release a script for noesis i am testing it out. [https://haruhi-chan4ever.deviantart.com/art/K-On-Houkago-Live-Noesis-Script-Download-719731521](url)

**Comment by lfdaryl** [comment]
Thanks a lot, for I want to get them for years.

**Comment by barquetin16** [comment]
Yes me too, the script is working very well, i am testing the ps3 version and some of them actually work.
Thank the creator of the script.
